requests_kerberos/kerberos_.py

```diff
@@ -209,14 +210,26 @@ def generate_request_header(self, response, host, is_preemptive=False):
             # w/ name-based HTTP hosting)
             kerb_host = self.hostname_override if self.hostname_override is not None else host
 
-            self._context[host] = ctx = spnego.client(
-                username=self.principal,
-                hostname=kerb_host,
-                service=self.service,
-                channel_bindings=self._cbts.get(host, None),
-                context_req=gssflags,
-                protocol="kerberos",
-            )
+            # to prevent failing compare headers test, don't pass password if None
+            if self.password:
```

Comment:
No need for the branching calls, just do `password=self.password` and it being `None` will work normally. I'm not sure what you mean by

> to prevent failing compare headers test, don't pass password if None

Reply:
it was because it was failing 14 tests, in this parts that look like:
```
    assert mock_client.call_args[1] == {
        "username": None,
        "hostname": "www.example.org",
        "service": "HTTP",
        "channel_bindings": None,
        "context_req": spnego.ContextReq.sequence_detect | spnego.ContextReq.mutual_auth,
        "protocol": "kerberos",
    }
```
https://github.com/requests/requests-kerberos/blob/3f672cf7d175d7c073e3a52eeb979e33930330ee/tests/test_requests_kerberos.py#L73C1-L80C6

since when I pass password even if it is none, it seems to add an explicit `"password": None` in the args, which fails the assert.

and I was too lazy to update all the tests hehe. 
but let me know if i should just add the additional args in the tests.

Reply:
I would fix the tests, no need to make the code more complicated than it needs to be.

Reply:
alright, i've updated the tests